test/end-to-end/cotisations/cotisations.page.js

```diff
@@ -0,0 +1,133 @@
+
+helpers.configure(chai);
+
+/* loading grid actions */
+const GA = require('../shared/GridAction');
+const GU = require('../shared/GridUtils');
+const FU = require('../shared/FormUtils');
+const components = require('../shared/components');
+
+class CotisationPage {
+  constructor() {
+    this.gridId = 'cotisation-grid';
+    this.cotisationGrid = element(by.id(this.gridId));
+    this.actionLinkColumn = 7;
+  }
+
+  /**
+   * send back the number of cotisations in the grid
+   */
+  getCotisationCount() {
+    return this.cotisationGrid
+      .element(by.css('.ui-grid-render-container-body'))
+      .all(by.repeater('(rowRenderIndex, row) in rowContainer.renderedRows track by $index'))
+      .count();
+  }
+
+  /**
+   * simulate the create cotisation button click to show the dialog of creation
+   */
+  createCotisation(cotisation) {
+    FU.buttons.create();
+    FU.input('CotisationModalCtrl.cotisation.label', cotisation.label);
+    FU.input('CotisationModalCtrl.cotisation.abbr', cotisation.abbr);
+
+    if (cotisation.is_employee) {
+      element(by.css('[name="is_employee"]')).click();
+    }
+
+    if (cotisation.is_percent) {
+      element(by.css('[name="is_percent"]')).click();
+    }
+
+    FU.input('CotisationModalCtrl.cotisation.value', cotisation.value);
+
+    components.accountSelect.set(cotisation.four_account_id, 'four_account_id');
+    components.accountSelect.set(cotisation.six_account_id, 'six_account_id');
+
+    FU.buttons.submit();
+    components.notification.hasSuccess();
+  }
+
+  /**
+   * block creation without the function name
+   */
+  errorOnCreateCotisation() {
+    FU.buttons.create();
+    FU.buttons.submit();
+    FU.validation.error('CotisationModalCtrl.cotisation.label');
+    FU.buttons.cancel();
+  }
+
+  /**
+   * simulate a click on the edit link of a function
+   */
+  editCotisation(label, updateCotisation) {
+    GU.getGridIndexesMatchingText(this.gridId, label)
+      .then(indices => {
+        const { rowIndex } = indices;
+        GA.clickOnMethod(rowIndex, this.actionLinkColumn, 'edit', this.gridId);
+        FU.input('CotisationModalCtrl.cotisation.label', updateCotisation.label);
+
+        FU.buttons.submit();
+        components.notification.hasSuccess();
+      });
+  }
+
+  /**
+   * simulate a click on the delete link of a function
+   */
+  deleteCotisation(label) {
+    GU.getGridIndexesMatchingText(this.gridId, label)
+      .then(indices => {
+        const { rowIndex } = indices;
+        GA.clickOnMethod(rowIndex, this.actionLinkColumn, 'delete', this.gridId);
+        components.modalAction.confirm();
+        components.notification.hasSuccess();
+      });
+  }
+
+  /**
+   * cancel deletion process
+   */
+  cancelDeleteCotisation(n) {
+    GA.clickOnMethod(n, this.actionLinkColumn, 'delete', this.gridId);
+    components.modalAction.dismiss();
+  }
+
+  /**
+   * forbid deletion of used function
+   */
+  errorOnDeleteCotisation(n) {
```

Comment:
Can you make this refer to a text instead of an index?  It will make the test much easier to modify in the future.